pkg/ipam/ipam_linux.go

```diff
@@ -113,7 +113,8 @@ func ConfigureIface(ifName string, res *current.Result) error {
 		return fmt.Errorf("failed to set %q UP: %v", ifName, err)
 	}
 
-	if v6gw != nil {
+	// Ignore settling when state is down since it would never settle
+	if v6gw != nil && link.Attrs().OperState == netlink.OperUp {
```

Comment:
We have to retrieve the link in `ip.SettleAddresses()`, shall we add this check there instead?